### PR TITLE
Fix TTL merge scheduling bug

### PR DESCRIPTION
**Pull request by lthaooo**
### Changelog category (leave one):

- Bug Fix (user-visible misbehaviour in official stable or prestable release)

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):

TTL merge may not be scheduled again if BackgroundExecutor is busy.
--merges_with_ttl_counter is increased in selectPartsToMerge()
--merge task will be ignored if BackgroundExecutor is busy
--merges_with_ttl_counter will not be decrease

As for StorageReplicatedMergeTree, this problem still exist in version 21.x, but not exist in version 22.x because merges_with_ttl_counter is increased after merge task submitted

#30645

> Information about CI checks: https://clickhouse.tech/docs/en/development/continuous-integration/

### CODE REVIEW

**Comment by lthaooo** [comment]
it looks like the check failure is not related to my changes

File: src/Storages/StorageMergeTree.cpp [hunk 1]
```diff
@@ -1162,7 +1162,9 @@ bool StorageMergeTree::scheduleDataProcessingJob(BackgroundJobsAssignee & assign
     {
         auto task = std::make_shared<MergePlainMergeTreeTask>(*this, metadata_snapshot, false, Names{}, merge_entry, share_lock, common_assignee_trigger);
         task->setCurrentTransaction(std::move(transaction_for_merge), std::move(txn));
-        assignee.scheduleMergeMutateTask(task);
+        bool scheduled = assignee.scheduleMergeMutateTask(task);
+        if (!scheduled && isTTLMergeType(merge_entry->future_part->merge_type))
+            getContext()->getMergeList().cancelMergeWithTTL();
         return true;
```
Comment (nikitamikhaylov) [inline src/Storages/StorageMergeTree.cpp:1168]: return scheduled?

**Comment by nikitamikhaylov** [comment]
This is only a subset of tests. Others are not executed due to TestsBugfixCheck (I am not sure). And this check failed because of CI database downtime..

**Comment by nikitamikhaylov** [comment]
@Mergifyio update

File: src/Storages/StorageMergeTree.cpp (hunk 1 above)
Comment (lthaooo) [reply src/Storages/StorageMergeTree.cpp]: it looks better

**Comment by nikitamikhaylov** [comment]
@Mergifyio update

**Pull request merged**